tests/test_utils.py

```diff
@@ -233,6 +233,30 @@ async def coroutine(i):
     assert results == list(range(10))
 
 
+@pytest.mark.asyncio
```

Comment:
Could we've additional tests, testing the behavior, when we submit more tasks than `max_concurrency` allows? Just that we don't introduce regressions, when someone changes something in `ConcurrentTasks` later on. 

`test_concurrent_runner_high_concurrency` addresses this partially, but I guess there are some more edge cases (max_concurrency being 10 -> submit 11 long running tasks -> cancel -> what happens with the 11th task?). Just that we've the intention how this works in a test for later reference. (or we expand the class doc string to describe these scenarios).

Reply:
>but I guess there are some more edge cases (max_concurrency being 10 -> submit 11 long running tasks -> cancel -> what happens with the 11th task?).

In this case, the first 10 jobs will be canclled, and the 11th task will be completed. I will add this test.

Reply:
This is more complicated than I thought. I'm trying to get the test working or prove that `ConcurrentTasks` doesn't work as expected.

Reply:
Updated: https://github.com/elastic/connectors-python/pull/1477/commits/d8ef6705005fc53b00fb32d4bcc48cf9346bbf60